tmt/steps/execute/__init__.py

```diff
@@ -255,6 +265,21 @@ def prepare_tests(self):
                 metadata_filename, tmt.utils.dict_to_yaml(test._metadata))
         return tests
 
+    def prepare_scripts(self, guest):
+        """
+        Prepare additional scripts for testing
+        """
+        # Install all scripts on guest
+        for script in self.scripts:
+            source = os.path.join(
+                SCRIPTS_SRC_DIR, os.path.basename(script.path))
+
+            for dest in [script.path] + script.aliases:
+                guest.push(
```

Comment:
Just adding some context you asked for at the hacking session.

The setup/teardown logic was present to be able to back-up the present files if there were any. Secondly, it was used to make the reboot setup faster (see https://github.com/teemtee/tmt/issues/926 ), since previously to achieve the backup, multiple `guest.execute()` calls were used which was slow. By using the setup/teardown script, we could just push these scripts (as a part of the main `guest.push()` before tests are run) and then execute a single script which was way better wrt speed.

However, since we agreed that backing up the present restraint files is not a priority (at least until someone reports that it is a problem for them), we should be able to get rid of this (which you did :+1: ).

On a related note, I am thinking if this iterative `guest.push()` won't be a bit slow as well since we are pushing one file after another (and there could be quite a lot of them possibly). But I can't think of anything much better, the only thing that really comes to my mind is again a setup-like script. The problem is that we need to get the files into `/usr/local/bin`. To achieve that, we would have to:
 - set up the scripts in tmt's workdir
 - the scripts are pushed to the guest using the `guest.push()` here https://github.com/teemtee/tmt/blob/main/tmt/steps/execute/internal.py#L332
 - move them from tmt's workdir to the appropriate place (this should ideally be one `guest.execute()` call for all scripts, so it would have to be some kind of a setup script, I guess)

What do you think?

Reply:
> On a related note, I am thinking if this iterative `guest.push()` won't be a bit slow as well since we are pushing one file after another (and there could be quite a lot of them possibly). But I can't think of anything much better, the only thing that really comes to my mind is again a setup-like script.

How about some kind of batch-like processing? If `Guest.push` could be modified to accept multiple sources, and as long as the destination is a directory, could a `rsync` command be created to push all of them at once into this directory? The loop here could then just collect source paths and call `Guest.push()` once for all of them (or per `dirpath(destination)`)

Reply:
I will take a look how slow it really is, but yeah, I could actually push it together even. But first I need to look @ coreos and what to do with containers

Reply:
@FrNecas so I looked at the speed, I do not see at all some slowness. So I am not adressing it, everything seems snappy to me. I really would like to avoid those setup like scripts. Also I do not see a reason for doing just one push for speed up, seems all quick to me. As there will be only handful of scripts, I would not spend more time on looking into this.

Reply:
I am resolving this, please reopen in case more discussion needed